### DISCUSSION TITLE
set two values through x-on:click

### DISCUSSION THREAD

**Discussion by snapey** [alpinejs/alpine#601] [4. General]
I'm looping over a collection in a view, and for each, I have a button that can open a modal.

The basics works fine, but what I could do with is setting two values with the @click, so I'm struggling to understand how the usual `x-on:click="open=21"`  could be extended

I have initialised alpine with x-data="{ open: 0, duration: 0}"  in the hope that I can dynamically set open and duration later.

**Comment by ryangjchandler** [comment]
Hi @snapey, you want to separate each statement with a semicolon, for example: `@click="open = 21; duration = 100"` as an example.

**Reply by snapey** [reply]
I'm sure I tried that. Grrr.  

Thanks

**Comment by markschellhas** [comment]
Hi @snapey, Did you ever find a solution to this? Facing the same issue at the moment.

**Reply by SimoTod** [reply]
https://github.com/alpinejs/alpine/discussions/601#discussioncomment-29983

**Reply by SimoTod** [reply]
See https://codepen.io/SimoTod/pen/qBoVZLW

**Reply by snapey** [reply]
used Ryan's tip above

**Comment by szekelygobe** [comment]
You need to us an array 
`x-data="{ open: 0, duration: 0}"`

`x-on:click="[open=2, duration=2]"`

**Reply by ekwoka** [reply]
this is not what you need to do.

It's actually a bad way to do it.

the expressions can be directly the function body, no tricks needed write it exactly as you might do

```js
function onClick($event) {
  // your code here
}
```

with the assumption your state is available as globals.

This is basically what Alpine does behind the scenes.

So just separating with `;` is enough

**Reply by szekelygobe** [reply]
Its not working for me with {open=2; duration=3}, it is working with [open=2, duration=2].

**Reply by ekwoka** [reply]
Yes, because it should JUST be `"open=2; duration=3"` no object, no array.

**Reply by szekelygobe** [reply]
Ok, that's working.
